prototyping/Search.py

```diff
@@ -0,0 +1,50 @@
+from MockArticle import Article
+
+class Search:
+    """
+    This class implements the functions to score every article 
+    by the amount of incidences between keywords and synonyms to articles
+
+    Attributes:
+        keywords: A list of keywords
+        synonyms: A list of synonyms for the keywords
+        articles: A list of articles
+        keywords_weight: The weight that a keyword has on the scorig algorithm
+            Optional: default argument as 1
+        synonyms_weight: The weight that a synonym has on the scoring algorithm
+            Optional: default argument as 1
+    """
+    def __init__(self, keywords, synonyms, articles, keywords_weight=1, synonyms_weight=1):
```

Comment:
In its final form, is the engine Search is going to handle converting raw query text => keywords and synonyms?

If so, I'd leave those out of the parameters for any class functions, since the caller of the functions shouldn't need to know that keyword extractoin & synonym lookup is happening behind the scenes.

If needed for prototype, you can hardcode an internal map of { raw query text : keyword lists } and { keywords : synonyms }.

Reply:
No, there will be a utility function that returns the keywords from the query text and another that returns a list of synonyms from the list of keywords. Should be the next pr.